src/compiler/cpp_generator.cc

```diff
@@ -547,6 +548,115 @@ void PrintHeaderClientMethod(grpc_generator::Printer* printer,
   }
 }
 
+void PrintHeaderClientMethodCallbackInterfacesStart(
+    grpc_generator::Printer* printer,
+    std::map<grpc::string, grpc::string>* vars) {
+  // This declares the interface for the callback-based API. The components
+  // are pure; even though this is new (post-1.0) API, it can be pure because
+  // it is an entirely new interface that happens to be scoped within
+  // StubInterface, not new additions to StubInterface itself
+  printer->Print("class experimental_async_interface {\n");
```

Comment:
Why not ExperimentalAsyncInterface?

Reply:
I think I might be ok with that. We've been using "experimental" in lower-case though, so let me see if we're ok with going with uppercase. I don't accidentally want to make this api just yet.

Reply:
Ok, if this isn't a source of strong feelings, I'd prefer to leave it alone. It's not very likely that more than 1 or 2 people will ever overload this interface in a different way, so I'd like to keep it as is. We can revisit when de-experimentalizing this.